### ISSUE TITLE
Horsepower not collected correctly

### ISSUE THREAD

**Issue by davidwhitedanecca** [ikennanwosu/Projects#1]
Where horsepower is above 99, only two digits are collected, which is why there is lots of data in the 10 to 30 range (these should be 100 to 300 horsepower).  I guess your machine learning decides that horsepower is not very useful and so uses engine capacity instead as there is presumably some correlation between these.

This code corrects this (line 169 in web_crawler.py):

               # vehicle horsepower: converts metric horsepower (PS) to mechanical horsepower (BHP)
                try:
                    hp = re.findall(r'\d+BHP|\d+PS', ' '.join(attrib))
                    if hp[0][-1]=='P':  #if the last char is 'P', i.e BHP
                        val = hp[0][0:-3]  #get everything except the last 3 chars 
                        unit = hp[0][-3:]  #get the last 3 chars (BHP)
                    else:
                        val = hp[0][0:-2]  #get everything except the last 2 chars 
                        unit = hp[0][-2:]  #get the last 3 chars (PS)

                    if unit == 'BHP':
                        bhp = val
                    else:
                        bhp = round(int(val) / 1.014)
                    bhp = pd.to_numeric(bhp)
                except IndexError:
                    bhp = None

                horse_power.append(bhp)

**Comment by ikennanwosu** [comment]
Well spotted Dave. Thank you very much. I will have to update the code, scrape the data all over again, train the models and compare the results. 

In the 'car_price_prediction.ipynb' file, Step 9.1.1, I have just learnt that it is bad practice to fit_transform() the X_test dataset as below.

def scale_data(data):
    scaler = StandardScaler()
    data = scaler.fit_transform(data)#.values.reshape(-1,1))
    return data

X_train = scale_data(X_train)
X_test = scale_data(X_test).


The reason for this can be found [here](https://stackoverflow.com/questions/48692500/fit-transform-on-training-data-and-transform-on-test-data/48692740). Really useful information.

**Comment by davidwhitedanecca** [comment]
No problem, I think you should also check step 3 where you remove duplicates, I think you are actually keeping a single version of each duplicate and losing all the unique values.  I think all you need to do is:

    auto.drop_duplicates

Also, I notice when you webcrawl the radius is set to 1500miles, so you probably get lots of duplicates as autotrader will choose the sequence of which cars to show you.  It might be faster to do a smaller radius, but I've not tested it.

**Comment by ikennanwosu** [comment]
You are indeed right. I was rather filtering out all the duplicates and using them for the prediction. Thanks for spotting this out. 
I will modify the script and run the models again. This is a great exercise for me I must say. The only to learn is picking a project and tackling all the problems as one gets along. I will try narrowing down the search area and see what I get.

Thank you very much for taking the time to go through the work.
